clients/src/main/java/org/apache/kafka/clients/telemetry/DefaultClientTelemetry.java

```diff
@@ -385,8 +387,12 @@ private int getPushIntervalMs(final GetTelemetrySubscriptionsResponseData data)
     }
 
     @Override
-    public void pushTelemetrySucceeded(PushTelemetryResponseData data) {
+    public void pushTelemetryReceived(PushTelemetryResponseData data) {
         log.debug("Successfully pushed telemetry; response: {}", data);
+        if (data.errorCode() != Errors.NONE.code()) {
```

Comment:
It seems to me that you don't actually have to `return` here. In fact, could you remove the `setState` call in `handlePushTelemetryResponseDataErrors` and just let it happen in `pushTelemetryReceived` as it does already?